### PR TITLE
feat: inject mask at reply

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes # mf-172

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/components/CompositionDialog/Composition.tsx [hunk 1]
```diff
@@ -67,14 +69,14 @@ export function Composition({ type = 'timeline', requireClipboardPermission }: P
...
     const UI = useRef<CompositionRef>(null)
-
     const networkSupport = activatedSocialNetworkUI.injection.newPostComposition?.supportedOutputTypes
     return (
         <DialogStackingProvider>
             <InjectedDialog keepMounted open={open} onClose={onClose} title={t('post_dialog__title')}>
                 <DialogContent>
                     <CompositionDialogUI
                         ref={UI}
+                        reply={reply}
```
Comment (Jack-Works) [inline packages/mask/src/components/CompositionDialog/Composition.tsx]: Do not pass down this props to CompositionUI, but handle it in `onSubmit_`.

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: packages/shared-base/src/Messages/Mask.ts [hunk 2]
```diff
@@ -62,6 +62,7 @@ export interface MaskEvents extends MaskSettingsEvents, MaskMobileOnlyEvents, Ma
     profilesChanged: UpdateEvent<ProfileIdentifier>[]
     relationsChanged: RelationChangedEvent[]
     pluginMinimalModeChanged: [id: string, newStatus: boolean]
+    message: AutoPasteFailedEvent
```
Comment (guanbinrui) [inline packages/shared-base/src/Messages/Mask.ts]: Use a proper name. This one is vague.

File: packages/mask/src/social-network-adaptor/twitter.com/utils/selector.ts [hunk 3]
```diff
@@ -95,12 +95,17 @@ export const sideBarProfileSelector: () => LiveSelector<E, true> = () =>
...
     querySelector<E>('[role="main"] :not(aside) > [role="progressbar"] ~ div [role="button"][aria-label]:nth-child(6)')
+
+export const isReply = () => !!location.pathname.match(/^\/\w+\/status\/\d+$/)
 export const postEditorDraftContentSelector = () => {
     if (location.pathname === '/compose/tweet') {
         return querySelector<HTMLDivElement>(
             '[contenteditable][aria-label][spellcheck],textarea[aria-label][spellcheck]',
         )
     }
+    if (isReply()) {
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/twitter.com/utils/selector.ts]: Here it is detecting whether a reply editor is used with the current location. Does it go to mismatch if the composition modal is opening? Like the pic below, both two editors exist.

image

File: packages/mask/src/social-network/types.ts [hunk 4]
```diff
@@ -192,9 +192,11 @@ export namespace SocialNetworkUI {
         export interface NativeCompositionAttachImageOptions {
             recover?: boolean
             relatedTextPayload?: string
+            reply?: boolean
```
Comment (guanbinrui) [inline packages/mask/src/social-network/types.ts]: Replying is only needed by Twitter. Please use a proper name if you need to change the option for all SNS.

File: packages/shared-base/src/Messages/Mask.ts [hunk 5]
```diff
@@ -80,6 +81,7 @@ export interface UpdateEvent<Data> {
 export interface CompositionRequest {
     readonly reason: 'timeline' | 'popup'
```
Comment (guanbinrui) [inline packages/shared-base/src/Messages/Mask.ts]: Can you keep using the `reason` field?

File: packages/mask/src/social-network-adaptor/twitter.com/utils/selector.ts (hunk 3 above)
Comment (guanbinrui) [reply packages/mask/src/social-network-adaptor/twitter.com/utils/selector.ts]: @lelenei

**Review by guanbinrui — APPROVED** [review]

**Comment by guanbinrui** [comment]
@DimensionDev/qa

**Comment by AinuX** [comment]
Checked

**Pull request merged**